### PR TITLE
PSP-8317: Lease and Licence Edit/View Fee Denomination forms

### PR DESCRIPTION
**Pull request by stairaku**

### CODE REVIEW

File: source/frontend/src/features/leases/add/AddLeaseYupSchema.ts [hunk 1]
```diff
@@ -114,6 +114,11 @@ export const AddLeaseYupSchema = Yup.object().shape({
   primaryArbitrationCity: Yup.string()
     .nullable()
     .max(200, 'Primary arbitration city must be at most ${max} characters'),
+  isPublicBenefit: Yup.string().nullable(),
```
Comment (devinleighsmith) [inline source/frontend/src/features/leases/add/AddLeaseYupSchema.ts:117]: As far as I can tell, these are defined as booleans in the frontend model.

File: source/frontend/src/features/leases/add/FeeDeterminationSubForm.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      >
+        <MediumTextArea field="feeDeterminationNote" />
+      </SectionField>
+    </Section>
+  );
+};
+
+export default FeeDeterminationSubForm;
+
+const MediumTextArea = styled(TextArea)`
```
Comment (devinleighsmith) [inline source/frontend/src/features/leases/add/FeeDeterminationSubForm.tsx:63]: Why is this necessary? I'm surprised that you needed to manually specify the height here.

File: source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+          {
+            ...getEmptyPropertyLease(),
+            property: {
+              ...mockApiProperty,
+              areaUnit: toTypeCode('test'),
+              landArea: 123,
+            },
+          },
+        ],
+        amount: 1,
```
Comment (devinleighsmith) [inline source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx:73]: when writing tests, please try and keep the test data as minimal as "focused" as possible on the parameters necessary to evaluate the component under test. For example, for this test, a large number of these fields will not have an effect on what is rendered, and can be removed.

File: source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    } = setup({
+      lease: {
+        ...getEmptyLease(),
+        isFinancialGain: true,
+      },
+    });
+    expect(getByDisplayValue('Yes')).toBeVisible();
+  });
+
+  it('renders the suggested Fee field', () => {
```
Comment (devinleighsmith) [inline source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx:115]: since there is a calculation, I'd actually rather see more tests for this then the other fields. In general we don't need to write one test for field as that doesn't scale, but we want to write tests for non-trivial logic or business rules.

Comment (devinleighsmith) [reply source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx:115]: https://confluence.th.gov.bc.ca/display/PIMS/Best+Practices#BestPractices-UnitTests

File: source/frontend/src/features/mapSideBar/acquisition/tabs/compensation/detail/CompensationRequisitionDetailView.test.tsx [hunk 5]
```diff
@@ -31,7 +31,9 @@ describe('Compensation Detail View Component', () => {
       <CompensationRequisitionDetailView
         acquisitionFile={renderOptions?.props?.acquisitionFile ?? mockAcquisitionFileResponse()}
         compensation={renderOptions?.props?.compensation ?? getMockApiDefaultCompensation()}
-        compensationProperties={renderOptions?.props?.compensationProperties ?? getMockCompensationPropertiesReq()}
+        compensationProperties={
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/acquisition/tabs/compensation/detail/CompensationRequisitionDetailView.test.tsx:34]: ?

**Comment by devinleighsmith** [comment]
@stairaku please ask Julian what he means by this:

(from confluence)
image
I don't see this in the implementation, but I'm also not sure where the above should be.

File: source/frontend/src/features/leases/add/AddLeaseYupSchema.ts (hunk 1 above)
Comment (stairaku) [reply source/frontend/src/features/leases/add/AddLeaseYupSchema.ts:117]: Taken out the nullable validation on the yup schema since is not needed.

File: source/frontend/src/features/leases/add/FeeDeterminationSubForm.tsx (hunk 2 above)
Comment (stairaku) [reply source/frontend/src/features/leases/add/FeeDeterminationSubForm.tsx]: Changed this since height is not needed

File: source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx (hunk 3 above)
Comment (stairaku) [reply source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx]: Reduced the amount of variable required for unit testing.

File: source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx (hunk 4 above)
Comment (stairaku) [reply source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx]: I deleted the fields tests and increase the number of unit tests regarding the calculation of the suggested fee.

File: source/frontend/src/features/mapSideBar/acquisition/tabs/compensation/detail/CompensationRequisitionDetailView.test.tsx (hunk 5 above)
Comment (stairaku) [reply source/frontend/src/features/mapSideBar/acquisition/tabs/compensation/detail/CompensationRequisitionDetailView.test.tsx]: I think this is just a lint thing. I no longer see this file on my PR update.

**Review by devinleighsmith — APPROVED** [review]
